src/main/java/org/shanerx/tradeshop/utils/BukkitVersion.java

```diff
@@ -0,0 +1,152 @@
+ * any damages of any kind, whether be material or moral, which were
+ * caused by their contribution(s) to the project. See the full License for more information
+ */
+
+package org.shanerx.tradeshop.utils;
+
+import org.bukkit.Bukkit;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class BukkitVersion {
+	private final String VERSION = Bukkit.getBukkitVersion();
+	private Map<String, Integer> verMap;
+
+    public BukkitVersion() {
+		verMap = getVerMap();
+    }
+
+    public String toString() {
+		return getMajor() + "." + getMinor() + "." + getPatch();
+    }
+
+    public String getFullVersion() {
+		return VERSION;
+    }
+
+	private int getMajor() {
+		return verMap.get("major");
+    }
+
+	private int getMinor() {
+		return verMap.get("minor");
+    }
+
+	private int getPatch() {
+		return verMap.get("patch");
+	}
+
+	public boolean isBelow(int major, int minor) {
+		if (getMajor() < major) {
+			return true;
+		} else if (getMajor() == major) {
+			return getMinor() < minor;
+		}
+
+		return false;
+    }
+
+	public boolean isBelow(int major, int minor, int patch) {
+		if (getMajor() < major) {
+			return true;
+		} else if (getMajor() == major) {
+			if (getMinor() < minor) {
+				return true;
+			} else if (getMinor() == minor) {
+				return getPatch() < patch;
+			}
+		}
+
+		return false;
+	}
+
+	public boolean isAbove(int major, int minor) {
+		if (getMajor() > major) {
+			return true;
+		} else if (getMajor() == major) {
+			return getMinor() > minor;
+		}
+
+		return false;
+    }
+
+	public boolean isAbove(int major, int minor, int patch) {
+		if (getMajor() > major) {
+			return true;
+		} else if (getMajor() == major) {
+			if (getMinor() > minor) {
+				return true;
+			} else if (getMinor() == minor) {
+				return getPatch() > patch;
+			}
+		}
+
+		return false;
+	}
+
+	public boolean isEqual(int major, int minor) {
+		return getMajor() == major && getMinor() == minor;
+	}
+
+	public boolean isEqual(int major, int minor, int patch) {
+		return getMajor() == major && getMinor() == minor && getPatch() == getPatch();
+	}
+
+	public boolean isInt(String str) {
+		try {
+			Integer.parseInt(str);
+			return true;
+		} catch (Exception e) {
+			return false;
+		}
+	}
+
+	public Map<String, Integer> getVerMap() {
+		Pattern pat = Pattern.compile("(?!\\.)(\\d+(\\.\\d+)+)(?![\\d\\.])");
+		Matcher matcher = pat.matcher(VERSION);
+
+		String ver;
+		if (matcher.find()) {
+			ver = matcher.group();
+		} else {
+			return null;
+		}
+
+		String[] verSplit = ver.split("\\.");
+
+		int[] verInts = new int[3];
+
+		for (int i = 0; i < verSplit.length; i++) {
+			if (isInt(verSplit[i])) {
+				verInts[i] = Integer.parseInt(verSplit[i]);
+			}
+		}
+
+		Map<String, Integer> map = new HashMap<>();
+		map.put("major", verInts[0]);
+		map.put("minor", verInts[1]);
+		map.put("patch", verInts[2]);
+
+		return map;
+	}
+}
```

Comment:
Do we need this? After all, support for anything below 1.13 is being dropped.
And if we are thinking about the future, we can always add it back. After all, it seems you have removed all calls to this class and its members.

Reply:
It is being used. just not fully, #isBelow is what is checking our versions in the main file and if we need to check version for features later this gives us the ability